src/WebUI/Common/KubeSummary.tsx

```diff
@@ -112,11 +118,38 @@ export class KubeSummary extends BaseComponent<IKubeSummaryProps, IKubernetesCon
         );
     }
 
+    public componentDidMount(): void {
+        // This needs to be called after the data is loaded so that we can decide which object is selected as per the URL
+        setTimeout(() => {
+            this._updateStateFromHistory(queryString.parse(this._historyService.location.search));
+            this._historyUnlisten = this._historyService.listen(this._onHistoryChanged);            
+        }, 100);
+    }
+
     public componentWillUnmount(): void {
         this._selectionStore.removeChangedListener(this._onSelectionStoreChanged);
         this._workloadsStore.removeListener(WorkloadsEvents.DeploymentsFetchedEvent, this._setNamespaceOnDeploymentsFetched);
         this._workloadsStore.removeListener(WorkloadsEvents.WorkloadsFoundEvent, this._onDataFound);
         this._servicesStore.removeListener(ServicesEvents.ServicesFoundEvent, this._onDataFound);
+        this._historyUnlisten();
+    }
+
+    private _updateStateFromHistory = (routeValues: queryString.OutputParams): void => {
+        if (routeValues["type"] && routeValues["uid"])
+        {
+            const typeName: string = routeValues["type"] as string;
+            const objectId: string = routeValues["uid"] as string;
+            const selectedItem = this._objectFinder[typeName](objectId);
+            this.setState({selectedItemType: typeName, selectedItem: selectedItem, showSelectedItem: true});
+        }
+        else {
+            this.setState({selectedItemType: "", selectedItem: undefined, showSelectedItem: false});
+        }
+    };
+
+    private _onHistoryChanged = (location: Location, action: Action): void => {      
+        let routeValues: queryString.OutputParams = queryString.parse(location.search);
```

Comment:
const, also we should see what happens if parse fail

Reply:
will do